cmake-modules/AddTestCMocka.cmake

```diff
@@ -65,6 +65,11 @@ function(ADD_CMOCKA_TEST _TARGET_NAME)
 
     add_executable(${_TARGET_NAME} ${_add_cmocka_test_SOURCES})
 
+    # Suppress clobber warning for longjmp
+    if(CMAKE_C_COMPILER_ID MATCHES "GNU")
+      target_compile_options(${_TARGET_NAME} PRIVATE -Wno-clobbered)
+    endif()
+
     if (DEFINED _add_cmocka_test_COMPILE_OPTIONS)
```

Comment:
Unrelated to this PR, but @vhvb1989 what does `_add_cmocka_test_COMPILE_OPTIONS` do and used for? I don't see it defined anywhere.

https://github.com/Azure/azure-sdk-for-c/search?q=_add_cmocka_test_COMPILE_OPTIONS&unscoped_q=_add_cmocka_test_COMPILE_OPTIONS